csdump/csdump.c

```diff
@@ -0,0 +1,616 @@
+/* Capstone Disassembler Engine - Cross Platform */
+/* VoltagedDebunked <rusindanilo@gmail.com> 2025 */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <capstone/capstone.h>
+#include <errno.h>
+
+/* Platform-specific includes and definitions */
+#ifdef _WIN32
+#include <windows.h>
+#include <io.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#define PATH_SEPARATOR '\\'
+#define open _open
+#define close _close
+#define read _read
+#define lseek _lseek
+#define fstat _fstat
+#define stat _stat
+#ifdef _MSC_VER
+#define strtoull _strtoui64
+#define strcasecmp _stricmp
+#endif
+#else
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <strings.h>
+#define PATH_SEPARATOR '/'
+#ifndef strcasecmp
+#define strcasecmp strcasecmp
+#endif
+#endif
+
+/* Cross-platform getopt implementation for Windows */
+#ifdef _WIN32
+extern char *optarg;
+extern int optind, opterr, optopt;
+
+struct option {
+	const char *name;
+	int has_arg;
+	int *flag;
+	int val;
+};
+
+#define no_argument 0
+#define required_argument 1
+#define optional_argument 2
+
+int getopt_long(int argc, char *const argv[], const char *optstring,
+		const struct option *longopts, int *longindex);
+#else
+#include <getopt.h>
+#endif
+
+#define MAX_ARCH_NAME 16
+#define MAX_FILE_SIZE (1024 * 1024 * 100) /* 100MB limit for safety */
+
+typedef enum { BIN_RAW, BIN_ELF, BIN_MZ } binary_format_t;
+
+typedef struct {
+	csh handle;
+	cs_arch arch;
+	cs_mode mode;
+	const char *filename;
+	uint64_t base_addr;
+	size_t start_offset;
+	size_t limit;
+	int color;
+	int detail;
+	binary_format_t format;
+} csdump_opts_t;
+
+/* Cross-platform file reading structure */
+typedef struct {
+	uint8_t *data;
+	size_t size;
+#ifdef _WIN32
+	HANDLE file_handle;
+	HANDLE mapping_handle;
+#else
+	int fd;
+#endif
+} file_mapping_t;
+
+void usage(const char *prog)
+{
+	printf("Usage: %s [options] -i <file>\n", prog);
+	puts("Options:");
+	puts("  -a, --arch <arch>     Architecture (x86, arm, aarch64, mips, ppc, sparc, systemz, riscv, etc)");
+	puts("  -m, --mode <mode>     Mode (16, 32, 64, thumb, etc)");
+	puts("  -i, --input <file>    Input binary file");
+	puts("  -b, --base <addr>     Base address (default 0x0)");
+	puts("  -s, --start <offset>  Start offset in file");
+	puts("  -l, --limit <bytes>   Max bytes to disassemble");
+	puts("  -d, --detail          Enable detailed instruction info");
+	puts("      --color           Enable colorized output");
+	puts("  -h, --help            Show this help");
+	puts("");
+	puts("Supported architectures:");
+	puts("  x86, arm, aarch64, mips, ppc, sparc, systemz, xcore, m68k,");
+	puts("  tms320c64x, m680x, evm, mos65xx, wasm, bpf, riscv, sh,");
+	puts("  tricore, alpha, hppa, loongarch, xtensa, arc");
+}
+
+cs_arch parse_arch(const char *name)
```

Comment:
All those option parsing stuff is already implemented in cstool.
Please use those ones so we don't have code duplicates.